backend/src/main/java/eu/clarin/switchboard/resources/ToolsResource.java

```diff
@@ -36,14 +34,40 @@ public ToolsResource(ToolRegistry toolRegistry, ToolConfig toolConfig) {
     @Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
     public Response getTools(@QueryParam("mediatype") String mediatype,
                              @QueryParam("language") String language,
-                             @QueryParam("onlyProductionTools") String onlyProductionTools) {
+                             @QueryParam("onlyProductionTools") String onlyProductionTools,
```

Comment:
If have no concrete suggestion to make it better but this method has a pretty complicated control flow and there are many 'exit points'. I don't believe it is doing anything insanely complex though, so perhaps some rewriting/refactoring could help.